terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -39,16 +41,20 @@ def updated_terraform_file_content(file)
         reqs = dependency.requirements.zip(dependency.previous_requirements).
                reject { |new_req, old_req| new_req == old_req }
 
-        # Loop through each changed requirement and update the files
+        # Loop through each changed requirement and update the files and lockfile
         reqs.each do |new_req, old_req|
           raise "Bad req match" unless new_req[:file] == old_req[:file]
           next unless new_req.fetch(:file) == file.name
 
           case new_req[:source][:type]
           when "git"
             update_git_declaration(new_req, old_req, content, file.name)
-          when "registry", "provider"
+          when "registry" 
             update_registry_declaration(new_req, old_req, content)
+          when "provider"
+            update_registry_declaration(new_req, old_req, content)
+          when "lockfile"
+            update_lockfile_declaration(new_req, old_req, content)
```

Comment:
I wonder if we should refactor these methods to return the updated content instead of mutating the `content` argument, this might make it easier to reason about `update_lockfile_declaration`, e.g.

```
when "provider"
  content = update_registry_declaration(new_req, old_req, content)
when "lockfile"
  content = update_lockfile_declaration(new_req, old_req, content)
...
```